table/table_test.cc

```diff
@@ -252,9 +252,11 @@ class KeyConvertingIterator : public InternalIterator {
   Slice key() const override {
     assert(Valid());
     ParsedInternalKey parsed_key;
-    if (ParseInternalKey(iter_->key(), &parsed_key) != Status::OK()) {
-      status_ = Status::Corruption("malformed internal key");
-      return Slice("corrupted key");
+    Status pikStatus = ParseInternalKey(iter_->key(), &parsed_key);
+    if (!pikStatus.ok()) {
+      status_ =
+          Status::Corruption("malformed internal key. ", pikStatus.getState());
+      return Slice(pikStatus.getState());
```

Comment:
Is it a memory leak? pikStatus would be destroyed after key() returns.

Also Nit: I'm not sure why we would construct the key as "malformed internal key. Corrupted Keys. key=...."

Reply:
Removed the unnecessary prefix. Didn't understand why there would be a memory leak...